ts/features/walletV3/payment/components/WalletPaymentConfirmContent.tsx

```diff
@@ -0,0 +1,177 @@
+import React from "react";
+import * as O from "fp-ts/lib/Option";
+import { pipe } from "fp-ts/lib/function";
+import {
+  Body,
+  GradientScrollView,
+  LabelLink,
+  ListItemHeader,
+  ModuleCheckout,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import { useNavigation } from "@react-navigation/native";
+import { openAuthenticationSession } from "@pagopa/io-react-native-login-utils";
+import {
+  AppParamsList,
+  IOStackNavigationProp
+} from "../../../../navigation/params/AppParamsList";
+import { WalletPaymentRoutes } from "../navigation/routes";
+import {
+  WALLET_PAYMENT_TERMS_AND_CONDITIONS_URL,
+  getPaymentLogo
+} from "../../common/utils";
+import { format } from "../../../../utils/dates";
+import { capitalize } from "../../../../utils/strings";
+import { PaymentRequestsGetResponse } from "../../../../../definitions/pagopa/ecommerce/PaymentRequestsGetResponse";
+import { Bundle } from "../../../../../definitions/pagopa/ecommerce/Bundle";
+import { WalletInfo } from "../../../../../definitions/pagopa/walletv3/WalletInfo";
+import {
+  TypeEnum,
+  WalletInfoDetails,
+  WalletInfoDetails1,
+  WalletInfoDetails2,
+  WalletInfoDetails3
+} from "../../../../../definitions/pagopa/walletv3/WalletInfoDetails";
+import I18n from "../../../../i18n";
+
+import { formatNumberCurrencyCentsOrDefault } from "../../../idpay/common/utils/strings";
+import { WalletPaymentTotalAmount } from "./WalletPaymentTotalAmount";
+
+export type WalletPaymentConfirmContentProps = {
+  paymentMethodDetails: WalletInfoDetails;
+  selectedMethod: WalletInfo;
+  selectedPsp: Bundle;
+  paymentDetails: PaymentRequestsGetResponse;
+  isLoading?: boolean;
+  onConfirm?: () => void;
+};
+
+export const WalletPaymentConfirmContent = ({
+  paymentMethodDetails,
+  selectedPsp,
+  selectedMethod,
+  paymentDetails,
+  isLoading,
+  onConfirm
+}: WalletPaymentConfirmContentProps) => {
+  const navigation = useNavigation<IOStackNavigationProp<AppParamsList>>();
+
+  const taxFee = pipe(
+    selectedPsp.taxPayerFee,
+    O.fromNullable,
+    O.getOrElse(() => 0)
+  );
+
+  const paymentAmount = pipe(
+    paymentDetails.amount,
+    O.fromNullable,
+    O.getOrElse(() => 0)
+  );
+
+  const totalAmount = paymentAmount + taxFee;
```

Comment:
```suggestion
   const taxFee = selectedPsp.taxPayerFee ?? 0;

  const paymentAmount = paymentDetails.amount ?? 0;

  const totalAmount = +paymentAmount + taxFee;
```
while FP does look very cool, I think we can skip the verbosity here; what do you think?

Reply:
Addressed into 3843a46f0b541a3809c37104b71a24f3a568a10f